### PR TITLE
Adds tests for parameter restructuring and assignment

### LINKED ISSUE #470: [Solidity Support] Param assignments
### Problem

The Hedera network supports solidity as one of the smart contracts development languages.
Occasionally items have shown up that surfaced in-equivalence at the language layer.

### Solution

Add examples for assignments operations

- Add `/contracts/solidity/asignments`
- Cover the tasklist below

```[tasklist]
### Assignments Tasks
- [x] Destructuring multiple returns e.g. https://docs.soliditylang.org/en/latest/control-structures.html#destructuring-assignments-and-returning-multiple-values
- [x] Memory vs storage assignment e.g. https://docs.soliditylang.org/en/latest/control-structures.html#complications-for-arrays-and-structs
```

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by konstantinabl**
**Description**:

* Adds contract for destructuring return values
* Adds contract for assignment of reference types
* Adds test for abovementioned contracts

**Related issue(s)**:

Fixes #470

### CODE REVIEW

File: contracts/solidity/assignments/AssignmentReferenceTypes.sol [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    function testAssignmentOfReferenceTypes() external {
+        testChangeCopy(someArray);
+        testChangeReference(someArray);
+    }
+
+    function testChangeCopy(uint[5] memory y) internal pure {
+        y[2] = 8;
+    }
+
+    function testChangeReference(uint[5] storage y) internal {
```
Comment (natanasow) [inline contracts/solidity/assignments/AssignmentReferenceTypes.sol:17]: we can add `pure` modifier as well

File: contracts/solidity/assignments/AssignmentReferenceTypes.sol [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        y[3] = 10;
+    }
+
+    function getSomeArray() external view returns(uint[5] memory){
```
Comment (natanasow) [inline contracts/solidity/assignments/AssignmentReferenceTypes.sol:21]: we should add space between `)` and `{` as it is in other functions #conventions

File: contracts/solidity/assignments/AssignmentReferenceTypes.sol [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        return someArray;
+    }
+}
```
Comment (natanasow) [inline contracts/solidity/assignments/AssignmentReferenceTypes.sol:24]: new line at the end

File: contracts/solidity/assignments/DestructuringReturns.sol [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    function f() public pure returns (uint, bool, uint) {
+        return (7, true, 2);
+    }
+
+    function testDestructuredReturnParams() external pure returns(uint, bool, uint) {
+        (uint x, bool y, uint z) = f();
+
+        return (x, y, z);
+    }
+}
```
Comment (natanasow) [inline contracts/solidity/assignments/DestructuringReturns.sol:15]: new line

**Review by natanasow — APPROVED** [review]
LGTM; approved with some optional nits

File: contracts/solidity/assignments/AssignmentReferenceTypes.sol [hunk 5]
```diff
@@ -0,0 +1,25 @@
+//SPDX-License-Identifier: UNLICENSED
```
Comment (Nana-EC) [inline contracts/solidity/assignments/AssignmentReferenceTypes.sol:1]: We usually go for Apache-2.0 on the licenses

File: test/solidity/assignments/assignments.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
+const { expect } = require('chai')
+const { ethers } = require('hardhat')
+const Constants = require('../../constants')
+
+describe('Test assignments', function () {
```
Comment (Nana-EC) [inline test/solidity/assignments/assignments.js:5]: Add test tag

File: test/solidity/assignments/assignments.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
+const { expect } = require('chai')
```
Comment (Nana-EC) [inline test/solidity/assignments/assignments.js:1]: Add license header

File: test/solidity/assignments/assignments.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+  it('should verify destructuring works', async function () {
+    const result = await contractDesctructuring.testDestructuredReturnParams();
+    expect(result).to.deep.equal([ethers.BigNumber.from(7), true, ethers.BigNumber.from(2)])
+  })
+
+  it('should verify assignment of reference types', async function () {
+    await contractReferenceTypes.testAssignmentOfReferenceTypes();
+    const result = await contractReferenceTypes.getSomeArray();
+    expect(result).to.deep.equal([ethers.BigNumber.from(1), ethers.BigNumber.from(2), ethers.BigNumber.from(3),
```
Comment (Nana-EC) [inline test/solidity/assignments/assignments.js:26]: nit: please add a comment here for readers to understand the nuance of the test so teh expectations are clear.

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Looks good. Some suggestions

File: test/solidity/assignments/assignments.js (hunk 7 above)
Comment (konstantinabl) [reply test/solidity/assignments/assignments.js:1]: We dont have license headers in the other js files?

**Review by Nana-EC — APPROVED** [review]
LG
Missing js file license header

**Review by Nana-EC — DISMISSED** [review]

**Review by Nana-EC — DISMISSED** [review]

**Review by Nana-EC — DISMISSED** [review]

**Pull request merged**